### PR TITLE
Animated Emoji Support

### LINKED ISSUE #463: Support animated emoji
- `Emoji#animated?`
- `Emoji#mention` serializes animated `a` flag
- `Bot#parse_mention` regex supports animated `a` flag

### PR DESCRIPTION
**Pull request by PixeLInc**
Closes #463

### CODE REVIEW

**Comment by Daniel-Worrall** [comment]
Specs?

File: lib/discordrb/data.rb [hunk 1]
```diff
@@ -2548,20 +2548,25 @@ class Emoji
...
       @id = data['id'].nil? ? nil : data['id'].to_i
+      @animated = data['animated']
 
       process_roles(data['roles']) if server
     end
 
     # @return [String] the layout to mention it (or have it used) in a message
     def mention
-      "<:#{@name}:#{@id}>"
+      "<#{animated ? 'a' : ''}:#{@name}:#{@id}>"
```
Comment (z64) [inline lib/discordrb/data.rb]: Consistently use the readers here please

**Comment by PixeLInc** [comment]
There's your specs, NeKkA

File: lib/discordrb/data.rb [hunk 2]
```diff
@@ -2579,7 +2584,7 @@ def icon_url
 
     # The inspect method is overwritten to give more useful output
     def inspect
-      "<Emoji name=#{@name} id=#{@id}>"
+      "<Emoji name=#{@name} id=#{@id} animated=#{@animated}>"
```
Comment (z64) [inline lib/discordrb/data.rb]: Could you use the readers here too?

File: lib/discordrb/data.rb (hunk 1 above)
Comment (Daniel-Worrall) [inline lib/discordrb/data.rb]: Could you change `@name` and `@id` to `name` and `id` just as `animated` is.

Comment (z64) [inline lib/discordrb/data.rb]: Could you consistently use the readers here please?

File: lib/discordrb/data.rb (hunk 2 above)
Comment (z64) [inline lib/discordrb/data.rb]: ditto

File: lib/discordrb/data.rb [hunk 3]
```diff
@@ -2548,20 +2548,25 @@ class Emoji
     # @return [Array<Role>] roles this emoji is active for
     attr_reader :roles
 
+    # @return [Boolean] if the emoji is animated
```
Comment (z64) [inline lib/discordrb/data.rb]: Ruby doesn't have a `Boolean` class. Please just write `true, false` for the type

File: lib/discordrb/data.rb [hunk 4]
```diff
@@ -2566,25 +2566,25 @@ def initialize(data, bot, server)
 
     # @return [String] the layout to mention it (or have it used) in a message
     def mention
-      "<#{animated ? 'a' : ''}:#{@name}:#{@id}>"
+      "<#{animated ? 'a' : ''}:#{name}:#{id}>"
```
Comment (LikeLakers2) [inline lib/discordrb/data.rb]: `#{'a' if animated}`?

**Pull request merged**